FinniversKit/Sources/Components/FrontPageShelfHeaderView/FrontPageShelfHeaderView.swift

```diff
@@ -0,0 +1,61 @@
+import UIKit
+
+public class FrontPageShelfHeaderView: UICollectionReusableView {
+    public typealias ButtonAction = (()->())
+    
+    private lazy var titleLabel: UILabel = {
+        let label = Label(style: .title3Strong)
+        label.translatesAutoresizingMaskIntoConstraints = false
+        label.textAlignment = .left
+        label.numberOfLines = 1
+        return label
+    }()
+    
+    private lazy var button: UIButton = {
+        let button = Button(style: Button.Style.flat)
+        button.size = .normal
+        button.translatesAutoresizingMaskIntoConstraints = false
+        button.setContentHuggingPriority(.required, for: .horizontal)
+        return button
+    }()
+    
+    public required init?(coder: NSCoder) {
+        fatalError("init(coder: ) has not been implemented")
+    }
+    
+    public override init(frame: CGRect) {
+        super.init(frame: frame)
+    }
+    
+    private(set) var title: String = ""
+    private(set) var buttonTitle: String = ""
+    private(set) var buttonAction: ButtonAction?
+    
+    public func configureHeaderView(withTitle title: String, buttonTitle: String, buttonAction: @escaping ButtonAction) {
+        self.title = title
+        self.buttonTitle = buttonTitle
+        self.buttonAction = buttonAction
+        
+        setup()
```

Comment:
Can `setup` be in `init` instead? Now it looks like the subviews can be added twice+ if this view is reused 🤔